src/h5web/App.test.tsx

```diff
@@ -1,23 +1,134 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import {
+  render,
+  RenderResult,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
 import App from './App';
 import MockProvider from './providers/mock/MockProvider';
 
 const DOMAIN = 'bsa_002_000-integrate-sub';
 
-test('renders with root group selected', async () => {
-  const { findByText, findByRole } = render(
+function renderApp(): RenderResult {
+  return render(
     <MockProvider domain={DOMAIN}>
       <App />
     </MockProvider>
   );
+}
+
+test('renders with root group selected', async () => {
+  const { findByText, getByRole } = renderApp();
 
   const noVisText = await findByText('Nothing to visualize');
-  expect(noVisText).toBeInTheDocument();
+  expect(noVisText).toBeVisible();
 
-  const title = await findByRole('heading', { name: DOMAIN });
-  expect(title).toBeInTheDocument();
+  const title = getByRole('heading', { name: DOMAIN });
+  expect(title).toBeVisible();
 
-  const domainBtn = await findByRole('treeitem', { name: DOMAIN });
+  const domainBtn = getByRole('treeitem', { name: DOMAIN });
+  expect(domainBtn).toBeVisible();
   expect(domainBtn).toHaveAttribute('aria-selected', 'true');
 });
+
+test('switch between "display" and "inspect" modes', async () => {
+  const {
+    findByRole,
+    findByText,
+    getByRole,
+    queryByRole,
+    queryByText,
+  } = renderApp();
+
+  const inspectBtn = getByRole('tab', { name: 'Inspect' });
+  const displayBtn = getByRole('tab', { name: 'Display' });
+
+  // Switch to "inspect" mode
+  fireEvent.click(inspectBtn);
+
+  const noVisText1 = queryByText('Nothing to visualize');
+  expect(noVisText1).not.toBeInTheDocument();
+
+  const groupIdRow1 = await findByRole('row', { name: /Entity ID/ });
+  expect(groupIdRow1).toBeVisible();
+
+  // Switch back to "display" mode
+  fireEvent.click(displayBtn);
+
+  const groupIdRow2 = queryByRole('row', { name: /Entity ID/ });
+  expect(groupIdRow2).not.toBeInTheDocument();
+
+  const noVisText2 = await findByText('Nothing to visualize');
+  expect(noVisText2).toBeVisible();
+});
+
+test('toggle explorer sidebar', async () => {
+  const { findByRole, getByRole } = renderApp();
+
+  const domainBtn = await findByRole('treeitem', { name: DOMAIN });
+  const sidebarBtn = getByRole('button', {
+    name: 'Toggle explorer sidebar',
+  });
+
+  expect(domainBtn).toBeVisible();
+  expect(sidebarBtn).toHaveAttribute('aria-pressed', 'true');
+
+  fireEvent.click(sidebarBtn);
+  expect(domainBtn).not.toBeVisible();
+  expect(sidebarBtn).toHaveAttribute('aria-pressed', 'false');
+
+  fireEvent.click(sidebarBtn);
+  expect(domainBtn).toBeVisible();
+  expect(sidebarBtn).toHaveAttribute('aria-pressed', 'true');
+});
+
+test('navigate groups in explorer', async () => {
```

Comment:
Would it make sense/be possible to put this test in a file `Explorer.test.tsx` as it tests only `Explorer` features ?

Reply:
Yeah, I did consider it, but it would require rendering the `Explorer` component instead of the whole app. Conceptually, I feel like rendering the whole app and testing bits of it is safer and more valuable for now, at least until we start exporting the components.

Reply:
:ok_hand: